### ISSUE TITLE
typeof type guard doesn't narrow the type of array elements

### ISSUE THREAD

**Issue by nahuel** [Microsoft/TypeScript#19341]
```ts
const l = ["hello", 
           () => "world"]

// Compiles OK:
const o = l[0]
const s1 : string = typeof o === 'string' ? o : o()  // OK

// ERROR compiling:
const s2 : string = typeof l[0] === 'string' ? l[0] : l[0]()
// TS2349: Cannot invoke an expression whose type lacks a call signature. 
// Type 'string | (() => string)' has no compatible call signatures.
```

As you see, after the `typeof` type guard, `o` type is correctly narrowed from `string | (() => string)` to `string` or to `(() => string)`, but in the next line the `l[0]` type is not narrowed so a compiler error is raised.

**TypeScript Version:**  2.5.3

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/10530

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
